solidity/test/fixtures/bridge-full-deployment.ts

```diff
@@ -0,0 +1,212 @@
+import { deployments, ethers, helpers } from "hardhat"
```

Comment:
@nkuba What's your position on this? Should we reference random beacon and ecdsa deployment scripts or do it in one bundle like here? Given https://github.com/keep-network/tbtc-v2/issues/267 maybe we should start with `bridge-full-deployment.ts` and remove it as a part of the work on #267?

I think we generally need to come up with a consistent approach for testing in `tbtc-v2` given we have more and more deployment scripts added. It feels to me that the current approach is:
- Unit tests DO NOT use deployment scripts given that testing with a real `WalletRegistry` would be too complex. Instead, for Bridge unit tests, we always load the fixture from `fixtures/bridge.ts`.
- The only test that uses deployment scripts is `Bridge.Deployment.test.ts` - currently in progress in #242.
- We do need integration tests. @tomaszslabon's work in #268 is the first attempt. We need one, consistent approach to load the entire system that would be reused by all integration tests (see the question to @nkuba).

Does it make sense? cc @dimpar @lukasz-zimnoch

Reply:
I've just realized `fixtures/bridge.ts` uses `await deployments.fixture()` and uses Smock to deploy `WalletRegistry` fake at the address of already deployed `WalletRegistry` from the deployment script. (I am surprised it works at all.) This is kind of spooky if we think we want to add proxy to the mix BUT it also means `bridge-full-deployment.ts` is [probably not needed](https://github.com/keep-network/tbtc-v2/blob/87a36ea6393db95e1391736a84cb8721ce3087e7/solidity/deploy/05_deploy_bridge.ts#L12-L15) (?)

Reply:
> What's your position on this? Should we reference random beacon and ecdsa deployment scripts or do it in one bundle like here?

We should base the tests on the deployment scripts. This way we can verify the deployment script. I don't like duplicating the deployment here, as it's likely the scripts become inconsistent at some point.

> Unit tests DO NOT use deployment scripts given that testing with a real WalletRegistry would be too complex. Instead, for Bridge unit tests, we always load the fixture from fixtures/bridge.ts.

Incorrect. The tests that use `bridgeFixture` from `fixtures/bridge.ts` are using the deployment scripts.
https://github.com/keep-network/tbtc-v2/blob/87a36ea6393db95e1391736a84cb8721ce3087e7/solidity/test/fixtures/bridge.ts#L17
The `walletRegistry` is faked there.

> We do need integration tests. @tomaszslabon's work in 
Integration test for slashing gas usage #268 is the first attempt. We need one, consistent approach to load the entire system that would be reused by all integration tests (see the question to @nkuba).

We do need the integration (or system) test that will work on the full deployment of all the modules, without any stubs.